crates/matrix-sdk-crypto/src/store/memorystore.rs

```diff
@@ -486,6 +493,16 @@ impl CryptoStore for MemoryStore {
         Ok(self.backup_keys.read().await.to_owned())
     }
 
+    async fn load_dehydrated_device_pickle_key(&self) -> Result<Option<DehydratedDeviceKey>> {
+        Ok(self.dehydrated_device_pickle_key.read().await.to_owned())
+    }
+
+    async fn delete_dehydrated_device_pickle_key(&self) -> Result<()> {
+        let mut lock = self.dehydrated_device_pickle_key.write().await;
+        *lock = None;
+        Ok(())
+    }
```

Comment:
Maybe I'm missing something, but is this method used anywhere? It's part of the trait, but It's not exposed to users nor do we internally delete the key anywhere.

Would it make sense to delete it automagically when we create a new dehydrated device?

Reply:
Exposed now. Should be used by apps.
It could make sense if we changed a bit the API, currently create does not need the pickle key, it's upload_keys that use it. Not sure how it would work. Need to talk a bit with @uhoreg to see if can have a better integrated usage of pickle key handling